tests/integrationtests/tst_perfparser.cpp

```diff
@@ -172,6 +173,99 @@ private slots:
         m_capabilities = host.perfCapabilities();
     }
 
+    void testFileErrorHandling()
+    {
+        PerfParser parser(this);
+        QSignalSpy parsingFailedSpy(&parser, &PerfParser::parsingFailed);
+        QString message;
+
+        const auto notThereFile = QStringLiteral("not_here");
+        parser.initParserArgs(notThereFile);
+
+        QCOMPARE(parsingFailedSpy.count(), 1);
+        message = qvariant_cast<QString>(parsingFailedSpy.takeFirst().at(0));
+        QVERIFY(message.contains(notThereFile));
+        QVERIFY(message.contains(QLatin1String("does not exist")));
+
+        const auto parentDirs = QStringLiteral("../..");
+        parser.initParserArgs(parentDirs);
+        // note: initializing parser args reset the attached spy counter
+        QCOMPARE(parsingFailedSpy.count(), 1);
+        message = qvariant_cast<QString>(parsingFailedSpy.takeFirst().at(0));
+        QVERIFY(message.contains(parentDirs));
+        QVERIFY(message.contains(QLatin1String("is not a file")));
+
+        const auto noRead = QStringLiteral("no_r_possible");
+        const auto shell = QStandardPaths::findExecutable(QStringLiteral("bash"));
+        if (!shell.isEmpty()) {
+            QProcess::execute(
+                shell, {QLatin1String("-c"), QStringLiteral("rm -rf %1 && touch %1 && chmod a-r %1").arg(noRead)});
+            parser.initParserArgs(noRead);
+
+            QCOMPARE(parsingFailedSpy.count(), 1);
+            message = qvariant_cast<QString>(parsingFailedSpy.takeFirst().at(0));
+            QVERIFY(message.contains(noRead));
+            QVERIFY(message.contains(QLatin1String("not readable")));
+        }
+    }
+
+    void testFileContent()
+    {
+        Settings::instance()->setAppPath(
+            QFileInfo(QStandardPaths::findExecutable(QStringLiteral("true"))).dir().path());
+        Settings::instance()->setExtraLibPaths(QFINDTESTDATA("file_content"));
+        PerfParser parser(this);
+        QSignalSpy parsingFailedSpy(&parser, &PerfParser::parsingFailed);
+        QSignalSpy parsingFinishedSpy(&parser, &PerfParser::parsingFinished);
+        QString message;
+
+        // check pre-exported perfparser
+        const auto perfData = QFINDTESTDATA("file_content/true.perfparser");
```

Comment:
please make this a data driven test, i.e. use

```
void testFileContent_data()
{
    QTest::addColumn<QString>("file");
    QTest::addColumn<bool>(isValid);
    QTest::addRow("true.perfparser") << QFINDTESTDATA("file_content/true.perfparer");
    ...
}

void testFileContent()
{
    QFETCH(QString, perfFile);
    QFETCH(bool, isValid);
    ...
    QTRY_COMPARE(parsingFinishedSpy.count(), 1);
    QCOMPARE(parsingFailedSpy.count(), isValid ? 0 : 1);
    ...
}
```

Reply:
done (with an explicit timeout and the message instead of the boolean)